### PR TITLE
Add support for Elastic Stack 6.x

### LINKED ISSUE #32: remove support for kibana4 user
The default kibana4 role is gone as per 2.3

> 2016/05/19 11:53:35 Warning: The following roles [kibana4] are unknown. Make sure to add them to the [/etc/elasticsearch/shield/roles.yml] file. Nonetheless the user will still be associated with all specified roles

### LINKED ISSUE #151: With the release of Elasticsearch 6.0 today, when will the ARM templates support it?
Today is an exciting day with the release of Elasticsearch 6.0.  Can you tell us when you think the ARM templates will support it?

Thanks in advance.

### PR DESCRIPTION
**Pull request by russcam**
This PR adds support for the latest patch version in each minor of Elastic Stack 6.x.

- Remove the `kibana4_server` generated password and user. In 5.x+, Kibana uses the `kibana` built-in user. In 2.x, Kibana uses the `es_kibana` user.
- Add parameter to be able to specify `bootstrap,password` for configuring the built-in users
- Add parameter to be able to specify the password for the built-in `logstash_system` user in 5.2.0+
- Remove all sudo calls in the provisioning scripts; scripts run privileged by default
- Add mount options as parameter to disk set up and add `nofail` mount option
- Increase retries from 30 to 60 to ensure node is up
- comment out conf-enabled include in monit configuration to reduce noise in stderr
- explicitly set `path.logs` to `/var/log/elasticsearch`
- explicitly set `path.data` to `/var/lib/elasticsearch` for non-data nodes
- Disallow `Standard_A0` and `Standard_A1` as SKUs for Kibana. They have insufficient RAM for the bundle optimising process in 6.x
- Update the Ubuntu 14.04 SKU for Elasticsearch 2.x
- Always specify Content-Type: application/json in all curl requests when setting up Elasticsearch

Closes #151 
Closes #32

### CODE REVIEW

**Comment by russcam** [comment]
Branch links require updating to point back to master before merging

**Comment by Mpdreamz** [comment]
Took the liberty to amend to this PR @russcam not sure what happened to `ui.gif`, tried to overwrite my copy with an older version but it yields nothing.

File: build/tasks/patch-values.js [hunk 1]
```diff
@@ -26,7 +28,7 @@ var clientNodeValues = _.range(0, allowedValues.numberOfClientNodes + 1)
   .filter(function(i) { return i <= 12 || (i % 5) == 0; })
   .map(function (i) { return { "label" : i + "", value : i }});
 
-var allowedLocations = _(["ResourceGroup"]).concat(allowedValues.locations)
+var allowedLocations = _(["ResourceGroup"]).concat(allowedValues.locations);
```
[the remark this answers is not in the document]
Comment (russcam) [reply build/tasks/patch-values.js]: good catch

File: parameters/password.parameters.json [hunk 2]
```diff
@@ -29,9 +29,11 @@
   "authenticationType":{"value":"password"},
   "sshPublicKey":{"value":"<value-ignored>"},
   "adminPassword":{"value":""},
+  "securityBootstrapPassword":{"value":""},
   "securityAdminPassword":{"value":""},
   "securityReadPassword":{"value":""},
   "securityKibanaPassword":{"value":""},
+  "securityLogstashPassword":{"value":""},
   "location":{"value":"ResourceGroup"},
```
[the remark this answers is not in the document]
Comment (russcam) [reply parameters/password.parameters.json]: thanks

File: parameters/ssh.parameters.json [hunk 3]
```diff
@@ -29,9 +29,11 @@
   "authenticationType":{"value":"sshPublicKey"},
   "sshPublicKey":{"value":"ssh-rsa <redacted>"},
   "adminPassword":{"value":"<value-ignored>"},
+  "securityBootstrapPassword":{"value":""},
   "securityAdminPassword":{"value":""},
   "securityReadPassword":{"value":""},
   "securityKibanaPassword":{"value":""},
+  "securityLogstashPassword":{"value":""},
   "location":{"value":"ResourceGroup"},
```
[the remark this answers is not in the document]
Comment (russcam) [reply parameters/ssh.parameters.json]: thanks

**Comment by russcam** [comment]
> When i deploy I get the following from kibana:
>
> Login is currently disabled. Administrators should consult the Kibana logs for more details. Happy to share my deploy parameters.

This issue is now fixed with a95879f so I'm going to merge this into `master` now

**Pull request merged**